sql/catalyst/src/main/scala/org/apache/spark/sql/internal/StaticSQLConf.scala

```diff
@@ -36,9 +36,13 @@ object StaticSQLConf {
     .createWithDefault(Utils.resolveURI("spark-warehouse").toString)
 
   val CATALOG_IMPLEMENTATION = buildStaticConf("spark.sql.catalogImplementation")
+    .doc("Spark built-in implementation is in-memory and hive," +
+      "when set to yourImpl(can be any words), " +
+      "you need also provide following configuration to make everything going" +
+      "spark.sql.catalogImplementation.yourImpl.builder" +
+      "spark.sql.catalogImplementation.yourImpl.externalCatalog")
     .internal()
```

Comment:
Technically, this is `.internal()` configuration. This PR seems to propose make this external conf.

Reply:
I find that internal didn't effect anything. So I have removed .internal() in latest commit.